src/main/scala/dhall/Expr.scala

```diff
@@ -0,0 +1,183 @@
+package dhall
+
+sealed trait Expr[+S, +A] {
```

Comment:
If we are going with variance annotations, we should add those to the data constructor types too.